### PR TITLE
COL-1476, redirect to /asset if assetId found in URL during LTI launch

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/COL-1476

Review with `?w=1` for simple diff.  Doing param check on server-side (LTI launch) allows for test coverage.

### CODE REVIEW

File: src/components/assets/AssetCard.vue [hunk 1]
```diff
@@ -68,11 +68,15 @@ export default {
...
+    thumbnailUrl: undefined
   }),
+  created() {
+    this.thumbnailUrl = this.asset.thumbnailUrl
+  },
   methods: {
-    imgError(asset) {
-      asset.thumbnailUrl = this.imgNotFound
+    imgError() {
+      this.thumbnailUrl = this.imgNotFound
```
Comment (johncrossman) [inline src/components/assets/AssetCard.vue:78]: Modifying prop `asset` attribute was causing error.

File: squiggy/api/auth_controller.py [hunk 2]
```diff
@@ -142,66 +141,70 @@ def _fetch(key):
...
+            app.logger.info(f'Created user during LTI launch: canvas_user_id={canvas_user_id}')
+
+        # Asset page is bookmark-able
+        match = re.search('assetId=([0-9]+)', request.headers.get('Referer', ''))
+        asset_id = match and match.group(1)
+
+        path = f'/asset/{asset_id}' if asset_id else ('/assets' if is_asset_library else '/engage')
+        params = f'canvasApiDomain={canvas_api_domain}&canvasCourseId={canvas_course_id}'
+        app.logger.info(f'LTI launch redirect: {path}?{params}')
+        return user, f'{path}?{params}'
```
Comment (johncrossman) [inline squiggy/api/auth_controller.py:207]: The diff above is mostly indentation mod.

File: src/components/assets/AssetCard.vue (hunk 1 above)
Comment (pauline2k) [reply src/components/assets/AssetCard.vue:78]: @johncrossman Oh, that makes sense, but I think changing it this way will reintroduce the bug in https://jira.ets.berkeley.edu/jira/browse/COL-1517. The problem is that asset data isn't loaded yet when the `created()` hook is fired, so `this.thumbnailUrl` remains undefined and is never updated.

Comment (johncrossman) [reply src/components/assets/AssetCard.vue:78]: Ah. I think I can tweak such that `thumbnailUrl` is initialized in `data: () => ...` using asset prop.

File: src/components/assets/AssetCard.vue [hunk 3]
```diff
@@ -67,12 +67,15 @@ export default {
...
   },
-  data: () => ({
-    imgNotFound: require('@/assets/img-not-found.png')
-  }),
+  data: function() {
+    return {
+      imgNotFound: require('@/assets/img-not-found.png'),
+      thumbnailUrl: this.asset.thumbnailUrl
+    }
+  },
```
Comment (johncrossman) [inline src/components/assets/AssetCard.vue:75]: @pauline2k - Amended. You think this will work?

Comment (pauline2k) [reply src/components/assets/AssetCard.vue:75]: Let’s try it!

**Pull request merged**